scripts/make_flat_spectrum_eor.py

```diff
@@ -0,0 +1,135 @@
+    component type.
+
+    Parameters
+    ----------
+    variance: float
+        Variance of the signal, in Kelvin^2, at the reference channel.
+    nside: int
+        HEALPix NSIDE parameter. Must be a power of 2.
+    ref_chan: int
+        Frequency channel to set as reference, if using freqs.
+    ref_zbin: int
+        Redshift bin number to use as reference, if using redshifts.
+    redshifts: numpy.ndarray
+        Redshifts at which to generate maps.
+        Optional if freqs is provided.
+    freqs: numpy.ndarray
+        Frequencies in Hz. Overrides redshifts, setting them to be
+        the redshifts of the 21 cm line corresponding with these frequencies.
+        Optional if redshifts is provided.
+
+    Returns
+    -------
+    pyradiosky.SkyModel
+        A SkyModel instance corresponding a white noise-like EoR signal.
+
+    Notes
+    -----
+    Either redshifts or freqs must be provided.
+    The history string of the returned SkyModel gives the expected amplitude.
+    """
+    if freqs is not None:
+        redshifts = f21 / freqs - 1
+        ref_zbin = np.argsort(redshifts).tolist().index(ref_chan)
+    elif redshifts is None:
+        freqs = f21 / (redshifts + 1)
+    else:
+        raise ValueError("Either redshifts or freqs must be set.")
+
+    npix = 12 * nside ** 2
+    nfreqs = freqs.size
+    omega = 4 * np.pi / npix
+
+    # Make some noise.
+    stokes = np.zeros((4, npix, nfreqs))
+    stokes[0, :, :] = np.random.normal(0.0, np.sqrt(variance), (npix, nfreqs))
+
+    ref_z = redshifts[ref_zbin]
+    redshifts.sort()
+    ref_zbin = np.where(np.isclose(redshifts, ref_z))[0][0]
+
+    voxvols = np.zeros(nfreqs)
+    for zi in range(nfreqs - 1):
+        dz = redshifts[zi + 1] - redshifts[zi]
+
+        vol = (
+            Planck15.differential_comoving_volume(redshifts[zi]).to_value("Mpc^3/sr")
+            * dz
+            * omega
+        )
+        voxvols[zi] = vol
+    voxvols[
+        nfreqs - 1
+    ] = vol  # Assume the last redshift bin is the same width as the next-to-last.
+
+    scale = np.sqrt(voxvols / voxvols[ref_zbin])
+    stokes *= scale
+    stokes = np.swapaxes(stokes, 1, 2)  # Put npix in last place again.
+
+    if not isinstance(freqs, units.Quantity):
+        freqs *= units.Hz
+
+    # The true power spectrum amplitude is variance * reference voxel volume.
+    pspec_amp = variance * voxvols[ref_zbin]
+    history_string = (
+        f"Generated flat-spectrum model, with spectral amplitude {pspec_amp:.3f} "
+    )
+    history_string += r"K$^2$ Mpc$^3$"
+
+    return SkyModel(
+        freq_array=freqs,
+        hpx_inds=np.arange(npix),
+        spectral_type="full",
+        nside=nside,
+        stokes=stokes * units.K,
+        history=history_string,
+    )
+
+
+if __name__ == "__main__":
+    # When run as a script, construct a noiselike skymodel for the MWA frequency channels.
+
+    var = 9.0
+    nside = 32
```

Comment:
Rather than hard coding these, can we use argparse to allow them to be set on the command line? I'd actually probably not default them to anything so the user actually has to think about what they want.

Reply:
And maybe something similar for the start/stop frequencies, Nfreqs and channel width.

Reply:
Thanks for the suggestion! I've updated the script to use argparse.